Source/SwiftLintFramework/Rules/Lint/UnusedCaptureListRule.swift

```diff
@@ -0,0 +1,128 @@
+import Foundation
+import SourceKittenFramework
+
+public struct UnusedCaptureListRule: ASTRule, ConfigurationProviderRule, AutomaticTestableRule {
+    public var configuration = SeverityConfiguration(.warning)
+
+    public init() {}
+
+    public static var description = RuleDescription(
+        identifier: "unused_capture_list",
+        name: "Unused Capture List",
+        description: "Unused reference in a capture list should be removed.",
+        kind: .lint,
+        nonTriggeringExamples: [
+            """
+            [1, 2].map { [weak self] num in
+                self?.handle(num)
+            }
+            """,
+            """
+            let failure: Failure = { [weak self, unowned delegate = self.delegate!] foo in
+                delegate.handle(foo, self)
+            }
+            """,
+            "{ [foo] in foo.bar() }()",
+            "sizes.max().flatMap { [(offset: offset, size: $0)] } ?? []"
+        ],
+        triggeringExamples: [
+            """
+            [1, 2].map { [↓weak self] num in
+                print(num)
+            }
+            """,
+            """
+            let failure: Failure = { [weak self, ↓unowned delegate = self.delegate!] foo in
+                self?.handle(foo)
+            }
+            """,
+            """
+            let failure: Failure = { [↓weak self, ↓unowned delegate = self.delegate!] foo in
+                print(foo)
+            }
+            """,
+            "{ [↓foo] in _ }()"
+        ]
+    )
+
+    private let captureListRegex = regex("^\\{\\h*\\[([^\\]]+)\\].*\\bin\\b")
```

Comment:
any particular reason to use `\h` here? although not common, the capture list could be in a new line

Reply:
ah, good point.